build-tools/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -1675,7 +1675,8 @@ void configureHttpWait(WaitForHttpResource wait) {
         if (settings.containsKey("xpack.security.http.ssl.certificate")) {
             wait.setCertificateAuthorities(getConfigDir().resolve(settings.get("xpack.security.http.ssl.certificate").toString()).toFile());
         }
-        if (settings.containsKey("xpack.security.http.ssl.keystore.path")) {
+        if (settings.containsKey("xpack.security.http.ssl.keystore.path")
```

Comment:
Very optional but I wonder if we should just throw here instead. Otherwise we end up swallowing invalid settings and opaquely defaulting to using `certificate_authorities` over `keystore.path`.

Reply:
This is already kinda weird setting since we are setting the trust store == keystore. This only says don't default the truststore == keystore if you are using a certificate authority. I suspect that we really also not do set the trust store == keystore if the trust store is explicitly set either. However, I haven't tested.  If anything we shouldn't cross the streams that way and ensure usages set them both independently even if they are the same. But that is out of scope. I am OK with saying if you set a certificate authority we won't set your trust store == key store.